src/gui/qgsnewmemorylayerdialog.cpp

```diff
@@ -40,33 +40,9 @@ QgsVectorLayer *QgsNewMemoryLayerDialog::runAndCreateLayer( QWidget *parent )
 
   QgsWkbTypes::Type geometrytype = dialog.selectedType();
 
-  QString geomType;
-  switch ( geometrytype )
-  {
-    case QgsWkbTypes::Point:
-      geomType = QStringLiteral( "point" );
-      break;
-    case QgsWkbTypes::LineString:
-      geomType = QStringLiteral( "linestring" );
-      break;
-    case QgsWkbTypes::Polygon:
-      geomType = QStringLiteral( "polygon" );
-      break;
-    case QgsWkbTypes::MultiPoint:
-      geomType = QStringLiteral( "multipoint" );
-      break;
-    case QgsWkbTypes::MultiLineString:
-      geomType = QStringLiteral( "multilinestring" );
-      break;
-    case QgsWkbTypes::MultiPolygon:
-      geomType = QStringLiteral( "multipolygon" );
-      break;
-    case QgsWkbTypes::NoGeometry:
-      geomType = QStringLiteral( "none" );
-      break;
-    default:
-      geomType = QStringLiteral( "point" );
-  }
+  QString geomType = QgsWkbTypes::displayString( geometrytype );
+  if ( geomType.isNull() )
```

Comment:
I guess that's slightly safer. If displayString is not finding the type then we'll have much bigger issues than this :)